analyses/snv-callers/scripts/01-calculate_vaf_tmb.R

```diff
@@ -0,0 +1,294 @@
+              'OpenPBTA-analysis') to annotation object RDS file to be analyzed.",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--bed_wgs", type = "character", default = "none",
+    help = "File path that specifies the caller-specific
+                BED regions file. Assumes from top directory, 'OpenPBTA-analysis'",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--bed_wxs", type = "character", default = "none",
+    help = "File path that specifies the WXS BED regions file. Assumes
+              from top directory, 'OpenPBTA-analysis'",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--overwrite", action = "store_true",
+    default = FALSE, help = "If TRUE, will overwrite any files of
+              the same name. Default is FALSE",
+    metavar = "character"
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+########### Check that the files we need are in the paths specified ############
+needed_files <- c(opt$maf, opt$metadata, opt$bed_wgs, opt$bed_wxs, opt$annot_rds,
+                  opt$cosmic)
+
+# Add root directory to the file paths
+needed_files <- file.path(root_dir, needed_files)
+
+# Get list of which files were found
+files_found <- file.exists(needed_files)
+
+# Report error if any of them aren't found
+if (!all(files_found)) {
+  stop(paste("\n Could not find needed file(s):",
+    needed_files[which(!files_found)],
+    "Check your options and set up.",
+    sep = "\n"
+  ))
+}
+
+################## Create output directories for this caller ##################
+# Caller specific results directory path
+caller_results_dir <- file.path(root_dir, opt$output)
+
+# Make caller specific results folder
+if (!dir.exists(caller_results_dir)) {
+  dir.create(caller_results_dir, recursive = TRUE)
+}
+
+####################### File paths for files we will create ####################
+vaf_file <- file.path(caller_results_dir, paste0(opt$label, "_vaf.tsv"))
+region_annot_file <- file.path(caller_results_dir, paste0(opt$label, "_region.tsv"))
+tmb_file <- file.path(caller_results_dir, paste0(opt$label, "_tmb.tsv"))
+
+# Declare metadata file name for this caller
+metadata_file <- file.path(
+  caller_results_dir,
+  paste0(opt$label, "_metadata_filtered.tsv")
+)
+
+##################### Check for files if overwrite is FALSE ####################
+# If overwrite is set to FALSE, check if these exist before continuing
+if (!opt$overwrite) {
+  # Make a list of the output files
+  output_files <- c(vaf_file, region_annot_file, tmb_file)
+
+  # Find out which of these exist
+  existing_files <- file.exists(output_files)
+
+  # If all files exist; stop
+  if (all(existing_files)) {
+    stop(cat(
+      "Stopping; --overwrite is not being used and all output files already exist: \n",
+      vaf_file, "\n",
+      region_annot_file, "\n",
+      tmb_file
+    ))
+  }
+  # If some files exist, print a warning:
```

Comment:
Good idea. Of course, overwriting should be a warning sometimes too... but we can't save everyone!